### ISSUE TITLE
CoqIDE completion does not work on gtk3

### ISSUE THREAD

**Issue by eponier** [coq/coq#9945]
#### Description of the problem
When we type a word that Coq recognizes, it tries to open the completion window, but that makes the main Coq window blink and the following error message is printed:
```
Gdk-Message: [...]: Window [...] is a temporary window without parent, application will not be able to position it on screen.
```

#### Coq Version

master

**Comment by eponier** [comment]
On a Fedora 29 with Gnome 3, if that helps.

**Comment by SkySkimmer** [comment]
Can't reproduce on debian testing/KDE. Completion looks fine to me.

**Comment by eponier** [comment]
Apparently, this is a Wayland-specific error.
See, for instance, https://bugs.eclipse.org/bugs/show_bug.cgi?id=532941.
If I use `GDK_BACKEND=x11`, the popup appears, but at strange positions.
Apparently, the solution is to systematically attach a parent to the popups.

**Comment by ppedrot** [comment]
I have short-term plans to replace the homemade implementation of completion by the native sourceview one. If this is considered a bugfix, it can probably be  backported into 8.10.

**Comment by herbelin** [comment]
@pedrot, here are other issues with the completion popup on the lablgtk3 version.

I have strange behavior with completion. Sometimes, I just see a thick line instead of the popup. Sometimes, I see nothing at all. It seems to be a synchronization problem, depending on the speed of typing. The size and position of the popup is also sometimes strange. If I type `eq_`, I have only 2 and a half visible choices and the rest visible only by scrolling. If I type `and_`, I have a window with 10 choices and no scrolling. 

Should we wait for the sourceview-based implementation or do you think that there is something simple to do as of now?

**Comment by ppedrot** [comment]
I have a branch that uses the GTK implementation, but it needs a bit of polishing still. I can submit it for others to test though.

**Comment by eponier** [comment]
@ppedrot Please do it!

**Comment by ppedrot** [comment]
https://github.com/ppedrot/coq/tree/gtk-ide-completion

**Comment by eponier** [comment]
Thanks!

**Comment by eponier** [comment]
Some feedback:
- it seems to work pretty well at first sight
- in particular, the current completion is sometimes too slow, which means that if I type very quickly for instance "assum", it gives completions for "assu" only (without the last letter), and this is painful in the long run; the new one does not have this problem, but this seems to come at the price of less reactivity. Indeed, the completion shows only if I stop typing and only after a small delay. Is this on purpose?
- the completion pop-up does not have any border, I think it would be better with some
- there is a lot of `invalid cast from 'GtkSourceCompletionItem' to 'GtkSourceCompletionProvider'` in the console

**Comment by ppedrot** [comment]
> Is this on purpose?

That's due to the default interactive delay set by GtkSourceview. I guess we can add an option to tweak it.

**Comment by ppedrot** [comment]
> the completion pop-up does not have any border, I think it would be better with some

This  is harder to fix, I don't think we have access to that much layout in the Gtk implementation

> there is a lot of invalid cast from 'GtkSourceCompletionItem' to 'GtkSourceCompletionProvider' in the console

This has been fixed in lablgtk3 beta6.

**Comment by Alizter** [comment]
I have noticed three issues with the completion:

It seems to pick up words I wouldn't normally consider, for example if you have a `statement.` the completion will show `statement` and `statement.`.

Another thing I have noticed is where the box appears. I would expect it to appear directly under where I am writing, or above it if there is no space. Many time I have seen it just cover where I am typing which is really distracting.

Finally, when typing something and then pressing backspace I would expect it go back too. But it stays on the suggested words.
